ompi/attribute/attribute_predefined.c

```diff
@@ -162,19 +162,18 @@ int ompi_attr_create_predefined(void)
     return ret;
 }
 
-
 int ompi_attr_free_predefined(void)
 {
     int ret;
 
     if (OMPI_SUCCESS != (ret = free_comm(MPI_TAG_UB)) ||
-        OMPI_SUCCESS != (ret = free_comm(MPI_HOST)) ||
-        OMPI_SUCCESS != (ret = free_comm(MPI_IO)) ||
+        OMPI_SUCCESS != (ret = free_comm(MPI_HOST)) || OMPI_SUCCESS != (ret = free_comm(MPI_IO)) ||
```

Comment:
ew, don't really love this change.  and same with the one a couple lines below.

Reply:
Looks like it does the right thing iff it can't fit all the statements on a single line. I see ```BreakBeforeBinaryOperators``` which would do:

```c
    if (OMPI_SUCCESS != (ret = free_comm(MPI_TAG_UB))||
        || OMPI_SUCCESS != (ret = free_comm(MPI_HOST))
        || OMPI_SUCCESS != (ret = free_comm(MPI_IO))
        || OMPI_SUCCESS != (ret = free_comm(MPI_HOST))
        || OMPI_SUCCESS != (ret = free_comm(MPI_IO)) ...
```

If that isn't acceptable I can look at other options. I can always open a PR on clang-tidy to get us something like ```BreakAfterBinaryOperators``` but it would not be available until clang 13.something.

Reply:
Apparently this is is the old anti-pattern but now considered the best practice. The break after the binary operator is considered the new anti-pattern. Interesting.

Reply:
I actually like the way it looks breaking before the operator. This way lines up the operators all in the same column which is easier to read.

Reply:
Hmm, that doesn't quite do it. It looks better but still may put multiple conditions on a single line.

Reply:
I always break after the operator, but I'm not sure I care that strongly.